website/docs/core-concepts/writing-code/README.md

```diff
@@ -115,34 +85,87 @@ You cannot simply write multi-line javascript between the mustache brackets; the
 }}
 ```
 
-## Multi-Line JavaScript
-
-Appsmith supports multi-line JS if it is an [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) (Immediately-Invoked Function Expression). The above invalid examples become valid if restructured and used as below.
-
-```javascript
-{{ 
-  (function() {
-      const array = QueryName.data;
-      const filterArray = array.filter((row) => row.id > 5);
-      return filterArray;
-   })()
-}}
-```
-
-```javascript
-{{ 
-  (function() {
-      if (Dropdown.selectedOptionValue === "1") {
-        return "Option 1";
-      } else {
-        return "Option 2";
-      }
-   })()
-}}
-```
-
-:::tip
-**Writing comments inside :**
-
-Note that you can write comments inside using JavaScript's multi-line comment syntax `/* */`, but single-line comments `//` are not supported inside.
-:::
+It's important to keep in mind that, when using JavaScript inside curly braces (`{{ }}`) the code is written in a way that's accurate, efficient, readable, and well-organized to make it easier to understand and maintain. 
```

Comment:
i think we can omit this line. If you need this, move to the section that talks about single line JS towards the end

Reply:
This is true for multi-line also. By using a `function()` we can add multiple lines of code in the mustache sign, so I kept it here. It helps to add closure and emphasis for single as well as multi-line code.